internal/util-complete/src/main/scala/sbt/internal/util/LineReader.scala

```diff
@@ -191,25 +182,75 @@ trait LineReader {
   def readLine(prompt: String, mask: Option[Char] = None): Option[String]
 }
 
-final class FullReader(
-    historyPath: Option[File],
-    complete: Parser[_],
-    val handleCONT: Boolean = JLine.HandleCONT,
-    val injectThreadSleep: Boolean = false
-) extends JLine {
-  protected[this] val reader = {
-    val cr = JLine.createReader(historyPath, in)
-    sbt.internal.util.complete.JLineCompletion.installCustomCompletor(cr, complete)
-    cr
+object LineReader {
+  // same as scala.tools.nsc.Properties.isEmacsShell
+  val isEmacsShell = sys.props.getOrElse("env.emacs", "") != ""
```

Comment:
a more reliable way nowadays to check if you're running inside emacs is no look for the environment variables `INSIDE_EMACS`

Reply:
Yeah. I just wanted to use the same method that scala REPL uses for symmetry. Maybe we should do a PR also there.